src/js/game/map_chunk_aggregate.js

```diff
@@ -0,0 +1,146 @@
+    }
+
+    /**
+     * Marks this chunk as dirty, rerendering all caches
+     * @param {number} chunkX
+     * @param {number} chunkY
+     */
+    markDirty(chunkX, chunkY) {
+        const relX = chunkX % globalConfig.chunkAggregateSize;
+        const relY = chunkY % globalConfig.chunkAggregateSize;
+        this.dirtyList[relY * globalConfig.chunkAggregateSize + relX] = true;
+        if (this.dirty) {
+            return;
+        }
+        this.dirty = true;
+        ++this.renderIteration;
+        this.renderKey = this.x + "/" + this.y + "@" + this.renderIteration;
+    }
+
+    /**
+     *
+     * @param {HTMLCanvasElement} canvas
+     * @param {CanvasRenderingContext2D} context
+     * @param {number} w
+     * @param {number} h
+     * @param {number} dpi
+     */
+    generateOverlayBuffer(canvas, context, w, h, dpi) {
+        const prevKey = this.x + "/" + this.y + "@" + (this.renderIteration - 1);
+        const prevBuffer = this.root.buffers.getForKeyOrNullNoUpdate({
+            key: "agg@" + this.root.currentLayer,
+            subKey: prevKey,
+        });
+
+        const overlaySize = globalConfig.mapChunkSize * CHUNK_OVERLAY_RES;
+        let onlyDirty = false;
+        if (prevBuffer) {
+            context.drawImage(prevBuffer, 0, 0);
+            onlyDirty = true;
+        }
+
+        for (let x = 0; x < globalConfig.chunkAggregateSize; x++) {
+            for (let y = 0; y < globalConfig.chunkAggregateSize; y++) {
+                if (onlyDirty && !this.dirtyList[globalConfig.chunkAggregateSize * y + x]) continue;
+                this.root.map
+                    .getChunk(
+                        this.x * globalConfig.chunkAggregateSize + x,
+                        this.y * globalConfig.chunkAggregateSize + y,
+                        true
+                    )
+                    .generateOverlayBuffer(
+                        context,
+                        overlaySize,
+                        overlaySize,
+                        x * overlaySize,
+                        y * overlaySize
+                    );
+            }
+        }
+
+        this.dirty = false;
+        this.dirtyList.fill(false);
+    }
+
+    /**
+     * Overlay
+     * @param {DrawParameters} parameters
+     */
+    drawOverlay(parameters) {
+        const aggregateOverlaySize =
+            globalConfig.mapChunkSize * globalConfig.chunkAggregateSize * CHUNK_OVERLAY_RES;
+        const sprite = this.root.buffers.getForKey({
+            key: "agg@" + this.root.currentLayer,
+            subKey: this.renderKey,
+            w: aggregateOverlaySize,
+            h: aggregateOverlaySize,
+            dpi: 1,
+            redrawMethod: this.generateOverlayBuffer.bind(this),
+        });
+
+        const dims = globalConfig.mapChunkWorldSize * globalConfig.chunkAggregateSize;
+        const extrude = 0.05;
+
+        // Draw chunk "pixel" art
+        parameters.context.imageSmoothingEnabled = false;
+        drawSpriteClipped({
```

Comment:
Worth noting there's a subtle visual change here: the individual chunk images aren't "extruded", just the whole. I didn't try playing on "potato mode" to see if that leads to visual issues

Reply:
shouldn't be an issue :)